sklearn/neighbors/_classification.py

```diff
@@ -277,7 +299,35 @@ def predict_proba(self, X):
             The class probabilities of the input samples. Classes are ordered
             by lexicographic order.
         """
+        check_is_fitted(self, "_fit_method")
         if self.weights == "uniform":
+            # TODO: systematize this mapping of metric for
+            # PairwiseDistancesReductions.
+            metric = _adjusted_metric(self.metric, self.p)
+            if self.metric == "precomputed":
+                X = _check_precomputed(X)
+
```

Comment:
WE need to call `self._validate_data` in the estimator public API (fit/predict/predict_proba/transform/kneighbors) before we call into to the `ArgKminClassMode.compute` Cython private API which should always be able to safely assume that the input validation has already be done at the estimator level.